src/backend/commands/copy.c

```diff
@@ -4473,6 +4484,50 @@ CopyFrom(CopyState cstate)
 	return processed;
 }
 
+/*
+ * A subroutine to update the insert counts in all partitions for AO relations.
+ */
+static void CopyFromUpdateAOInsertCountInPartitions(HTAB *partition_hash, 
+													HTAB *tupcounts_ht,
+													List *ao_segnos)
+{
+	HASH_SEQ_STATUS hash_seq_status;
+	ResultPartHashEntry *entry;
+
+	hash_seq_init(&hash_seq_status, partition_hash);
```

Comment:
shouldn't we use `hash_freeze` before `hash_seq_init`?

Reply:
Additional protection will be useful. Thanks.